File: 0204.计数质数/0204-计数质数.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n < 2:
            return 0
        arr = [True]*n
        index = 2
        while index < int(n**0.5)+1:
            if arr[index] == True:
                step = index
                cur = index + step
                while cur < n:
                    arr[cur] = False
                    cur += step
            index += 1
        return sum(arr[2:])

        
                    
        

File: 0204.计数质数/0204-计数质数.rs
impl Solution {
    pub fn count_primes(n: i32) -> i32 {
        if n < 3{
            return 0;
        } else {
            let mut ret:Vec<i32> = vec![1;n as usize];
            for i in 2..((n/2 +1) as usize) {
                if ret[i] == 1 {
                    let mut index = 2*(i as i32);
                    while index < n {
                        ret[index as usize] = 0;
                        index += i as i32;
                    }
                }
            }
            return ret.iter().sum::<i32>() - 2;
        }
        unreachable!();
        
    }
}